portal/src/graphql/adminapi/UserDetailsConnectedIdentities.tsx

```diff
@@ -49,39 +52,50 @@ interface UserDetailsConnectedIdentitiesProps {
   availableLoginIdIdentities: string[];
 }
 
-const identityTypes = ["email", "phone", "username"] as const;
-type IdentityType = typeof identityTypes[number];
+const loginIdIdentityTypes = ["email", "phone", "username"] as const;
+type LoginIDIdentityType = typeof loginIdIdentityTypes[number];
+type IdentityType = LoginIDIdentityType | "oauth";
 
-interface EmailIdentityListItem {
+interface OAuthIdentityListItem {
   id: string;
-  email: string;
+  providerType?: OAuthSSOProviderType;
+  name?: string;
   verified: boolean;
   connectedOn: string;
 }
 
+interface EmailIdentityListItem {
+  id: string;
+  email?: string;
+  verified: boolean;
+  addedOn: string;
+}
+
 interface PhoneIdentityListItem {
   id: string;
-  phone: string;
+  phone?: string;
   verified: boolean;
   addedOn: string;
 }
 
 interface UsernameIdentityListItem {
   id: string;
-  username: string;
+  username?: string;
   addedOn: string;
 }
 
 export interface IdentityLists {
```

Comment:
I suggest refactoring the list items into two interfaces:
```ts
interface LoginIDIdentityListItem {
  id: string;
  type: "login_id";
  key: "email" | "phone" | "username"
  value: string;
  verified: boolean;
  connectedOn: string;
}

interface OAuthIdentityListItem {
  id: string;
  type: "oauth";
  provider: string;
  name: string;
  verified: boolean;
  connectedOn: string;
}
```

Then, `IdentityListCell` can accept just a list item instead of a variety of fields.

Reply:
Nice idea, refactored to these two interfaces